### ISSUE TITLE
[Doc]: Incorrect formatting for examples in user examples for user_explain

### ISSUE THREAD

**Issue by Higgs32584** [matplotlib/matplotlib#25582]
### Documentation Link

https://matplotlib.org/stable/tutorials/toolkits/index.html

### Problem

All the code is commented out in all three downloadable python examples from toolkits. When you download the files, either the .py or the .ipynb, built from each separate .py file, all the code is commented out in each example.

https://matplotlib.org/stable/tutorials/toolkits/axes_grid.html#sphx-glr-tutorials-toolkits-axes-grid-py
https://matplotlib.org/stable/tutorials/toolkits/axisartist.html
https://matplotlib.org/stable/tutorials/toolkits/mplot3d.html

### Suggested improvement

The solution would be to go through these files and format them correctly so that the code in the files is runnable without having to reformat the files yourself. The paths of the downloadable files referenced on the website are listed below.
./matplotlib/galleries/users_explain/toolkits/axes_grid.py
./matplotlib/galleries/users_explain/toolkits/axisartist.py
./matplotlib/galleries/users_explain/toolkits/axisartist.py

**Comment by miniMaddy** [comment]
Hey @Higgs32584, I want to work on this issue. I had one doubt. Do I just uncomment the code in the files, or can I change the comments to make the code more understandable? A lot of it referencing images which is good for website but might be confusing in the actual python file.

**Comment by Higgs32584** [comment]
Yeah, so I am still trying to get my first GitHub commit as well, so I do not know the codebase too well. 


Right now, it looks like you have to uncomment the code on both sides, cause it is intersped with regular comments as well. 


Also, it looks like a lot of the example code won't run without it's proper imports, and some initial data objects are missing, so you will have to add those as well. Finally, you'll have to load the documentation in sphinx to make sure it is framed correctly. I tried doing this today, and it looks like loading the documentation can take a bit of time.


If you are confused, you can look at another downloadable example. It looks like someone forgot to finish writing it out. 

I would also wait until we get an admin to comment on this before you begin working on it.

**Comment by jklymak** [comment]
This Issue will be a bit confusing given the links above.  Those three files were moved, and are now served at https://matplotlib.org/devdocs/users/explain/toolkits/index.html

This code is somewhat confusing because it is not really written as a sphinx-gallery py file, but rather look to have been translated from bare rst and reformatted as python comments to be able to be run through sphinx gallery.  Note that _none_ of the code is executed in either sphinx-gallery nor using the plot directive, and that the figures are all made in the examples gallery, eg, https://matplotlib.org/stable/gallery/axes_grid1/demo_axes_rgb.html, and simply linked here.  

My suggestion would be that these get re-written as *.rst files, which will get rid of the download links. If people want the code downloads, they can click through to the individual gallery examples.  If we do this, then the index.rst will need to be made by hand.

**Comment by Higgs32584** [comment]
When I changed the .py files to .rst they show up like this. Where do I move the .rst files so that they render correctly?
![Screenshot 2023-03-31 134324](https://user-images.githubusercontent.com/55243596/229192571-75054e6f-0a96-4a37-8829-4d9429f4c2e4.png)


Also, how do we rewrite the index.rst page? Thank you
![Screenshot 2023-03-31 134304](https://user-images.githubusercontent.com/55243596/229192587-c3c373d8-f8f8-4aa8-91ac-c2c68ab13635.png)

**Comment by jklymak** [comment]
Did you build the gallery?  If not, then the figure directives will yield broken links ...

As for the index.rst it isn't great currently - I'd just add a toctree command to it (look for :toctree: in the rest of the codebase to see how to do this).

**Comment by Higgs32584** [comment]
No, I have not built another index. and I would create the index.rst here, correct?

C:\Users\mjhig\.vscode\bob\matplotlib\galleries\users_explain\toolkits

**Comment by Pritishchugh22** [comment]
Hi @jklymak 
Hope you are doing well.
I am writing this to express my interest in participating in the Google Season of Docs program, 2023. I am a final year student at IIT Mandi, pursuing Bachelors of Technology in Computer Science. I am doing minors in Data Science and have done a significant number of courses as well as projects in this domain.
I wish if you could guide me on how I can start contributing and stand out from other participants.
Could you please guide me on how to proceed. I would appreciate any advice or recommendations you may have regarding the project.

I am attaching my resume as well as the link to my github which contains all my projects.
Resume: https://drive.google.com/file/d/1HR2JnSVzFPm4Vr76g_m2Betl8mT85jZQ/view?usp=drivesdk
Github: [github.com/pritishchugh22](http://github.com/pritishchugh22)

Thank you for your time and consideration. I hope to hear from you soon.
Thanks and regards
Pritish

**Comment by rcomer** [comment]
@Pritishchugh22 please see out contributors’ guide here:
https://matplotlib.org/devdocs/devel/contributing.html
